server/catgenome/src/main/java/com/epam/catgenome/manager/heatmap/HeatmapManager.java

```diff
@@ -0,0 +1,390 @@
+        heatmap.setType(BiologicalDataItemResourceType.FILE);
+        heatmap.setFormat(BiologicalDataItemFormat.HEATMAP);
+        heatmap.setCreatedDate(new Date());
+        heatmap.setSource(path);
+        readHeatmap(heatmap);
+        byte[] content = FileUtils.readFileToByteArray(file);
+        byte[] labelAnnotation = readFileContent(heatmap.getLabelAnnotationPath(), heatmap, this::checkLabelAnnotation);
+        byte[] cellAnnotation = readFileContent(heatmap.getCellAnnotationPath(), heatmap, this::checkCellAnnotation);
+        byte[] rowTree = readFileContent(heatmap.getRowTreePath(), heatmap, h -> {});
+        byte[] columnTree = readFileContent(heatmap.getColumnTreePath(), heatmap, h -> {});
+        biologicalDataItemManager.createBiologicalDataItem(heatmap);
+        heatmap.setBioDataItemId(heatmap.getId());
+        return heatmapDao.saveHeatmap(heatmap,
+                content,
+                cellAnnotation,
+                labelAnnotation,
+                rowTree,
+                columnTree);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void updateLabelAnnotation(final long heatmapId, final String path) throws IOException {
+        File file = getFile(path);
+        Heatmap heatmap = getHeatmap(heatmapId);
+        heatmap.setLabelAnnotationPath(path);
+        checkLabelAnnotation(heatmap);
+        heatmapDao.updateLabelAnnotation(heatmapId, FileUtils.readFileToByteArray(file), path);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void updateCellAnnotation(final long heatmapId, final String path) throws IOException {
+        File file = getFile(path);
+        Heatmap heatmap = getHeatmap(heatmapId);
+        heatmap.setCellAnnotationPath(path);
+        checkCellAnnotation(heatmap);
+        heatmapDao.updateCellAnnotation(heatmapId, FileUtils.readFileToByteArray(file), path);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void deleteHeatmap(final long heatmapId) {
+        Heatmap heatmap = getHeatmap(heatmapId);
+        heatmapDao.deleteHeatmap(heatmapId);
+        biologicalDataItemManager.deleteBiologicalDataItem(heatmap.getBioDataItemId());
+    }
+
+    public Heatmap loadHeatmap(final long heatmapId) {
+        return heatmapDao.loadHeatmap(heatmapId);
+    }
+
+    public List<List<Map<?, String>>> getContent(long heatmapId) throws IOException {
+        Heatmap heatmap = getHeatmap(heatmapId);
+        final String separator = getSeparator(heatmap.getPath());
+        try (InputStream heatmapIS = heatmapDao.loadHeatmapContent(heatmapId);
+                InputStream annotationIS = heatmapDao.loadCellAnnotation(heatmapId)) {
+            return getAnnotatedContent(heatmapIS, annotationIS, heatmap.getCellValueType(), separator);
+        }
+    }
+
+    public Map<String, String> getLabelAnnotation(final long heatmapId) throws IOException {
+        Heatmap heatmap = heatmapDao.loadHeatmap(heatmapId);
```

Comment:
If tree exists, but no annotation is specified an error `error.unsupported.heatmap.file.extension` is thrown, let's return empty response instead